### PR TITLE
DEV: name without dot

### PR DESCRIPTION
**Pull request by licode**
Exclude dot from key names.

### CODE REVIEW

**Comment by licode** [comment]
address https://github.com/NSLS-II/bluesky/issues/572

File: event_model/schemas/event_descriptor.json [hunk 1]
```diff
@@ -62,6 +62,9 @@
                 "description": "Creation time of the document as unix epoch time."
             }
         },
+        "patternProperties": {
+            "^([^\.]+)$": { "type": ["object", "integer", "string", "number", "array"] }},
```
Comment (licode) [inline event_model/schemas/event_descriptor.json]: yes. it can be empty dict.

**Comment by danielballan** [comment]
Nice!

I am pretty sure we don't want to set additionalProperties to false. Doesn't that disallow user-specified metadata? The event schema does block additional properties (by design) but all of the other documents are "dynamic," allowing arbitrary extra fields.

**Comment by tacaswell** [comment]
I think if you do not set 'additionalProperties' to False, then the format checking (effectively) does not apply.  If I understand it correctly, it is 'may there be properties that are not permitted by some other rule'.

@licode Does this apply recursively? ex, 
```
doc = {...,  # mandatory start doc stuff
       'valid_key': {'invalid.subkey': 1}}
```
should fail.

**Comment by licode** [comment]
Now this can check name recursively.

**Comment by danielballan** [comment]
Nice!

I wrote a test for bluesky verifying that names with dots raise validation errors (when run against this branch of event_model). I found some bugs -- this actually isn't valid JSON. One way to test that is simply to try importing the event_model package.

I opened up a PR against your branch. With those fixes, my bluesky test passes, so this should be good to merge. I'll push my bluesky test once this is merged.

**Comment by licode** [comment]
Neat. I just merged that.

**Comment by danielballan** [comment]
Great. Thanks again for diving into jsonschema, @licode.

**Pull request merged**